docs/source/howto/aforms.rst

```diff
@@ -0,0 +1,56 @@
+.. _aforms:
+
+How to extract AcroForm fields from a PDF using Python and PDFMiner
+********************************
+
+Before you start, make sure you have :ref:`installed pdfminer.six<install>`.
+
+The second thing you need is a PDF with AcroForms (as found in PDF files with fillable forms or multiple choices).
+
+It won't work for XFA forms.
+
+.. code-block:: python
+
+    from pdfminer.pdfparser import PDFParser
+    from pdfminer.pdfdocument import PDFDocument
+    from pdfminer.pdftypes import resolve1, PDFObjRef
+    from pdfminer.psparser import PSLiteral, PSKeyword
+    from pdfminer import utils
+    
+    data = {}
+    
+    with open(file_path, 'rb') as fp:
+        parser = PDFParser(fp)
+        
+        doc = PDFDocument(parser)
+        res = resolve1(doc.catalog)
+
+        if 'AcroForm' in res:
+            fields = resolve1(doc.catalog['AcroForm'])['Fields']
+
+            for f in fields:
+                field = resolve1(f)
+                name, value = field.get('T'), field.get('V')
+
+                # decode name
+                name = name.decode()
+
+                if value:
+
+                    # resolve indirect obj
+                    while isinstance(value, PDFObjRef):
+                        value = resolve1(value)
```

Comment:
I think this can be replaced by just `value = resolve1(value)`. The `resolve1` method already has a while loop in it.

Reply:
You're right, no sense repeating both the while loop and the isinstance check.